controller/controller.go

```diff
@@ -246,7 +246,10 @@ func (c *Controller) delete(obj Object) {
 }
 
 func (c *Controller) propagate(resourceEvents []ResourceEvent) {
-	topology := c.topology.Build(c.cache.List())
+	topology, err := c.topology.Build(c.cache.List())
+	if err != nil {
+		panic(err)
+	}
```

Comment:
I am unsure if this is the correct place to handle the error or if it should be passed up the stack more.

Reply:
Yep. Not sure about panicking here. I can see how handling the error instead may look like a better option. However, if a topology is set to be a DAG, we probably want the controller to scream it as soon as possible. WDYT?

Reply:
I guess if we're gonna give it the option to enable/disable the validation, we may not panic here. Or, rather, the option needs to leak to the controller, otherwise it's only optional if you use the topology standalone, but not the controller. Make sense?